bsp/boards/board.h

```diff
@@ -22,16 +22,50 @@ typedef enum {
    KICK_SCHEDULER        = 1,
 } kick_scheduler_t;
 
+typedef enum {
+   TIME_MS,
+   TIME_TICS,
+} time_type_t;
+
 //=========================== typedef =========================================
+typedef struct {
+    uint16_t PORT_SLOTDURATION;
+    uint16_t PORT_TsSlotDuration;
+
+    // execution speed related
+    // also implementation related (e.g. SoC radio or spi-connected radio because of the time to load a packet to/from radio)
+    uint16_t PORT_maxTxDataPrepare;
+    uint16_t PORT_maxRxAckPrepare;
+    uint16_t PORT_maxRxDataPrepare;
+    uint16_t PORT_maxTxAckPrepare;
+
+    // radio speed related
+    // also implementation related (because the time to execute the Tx/Rx command is highly dependent on the radio AND the configuration)
+    uint16_t PORT_delayTx;                       
+    uint16_t PORT_delayRx;
+} slot_board_vars_t; //board specific slot vars
+
+// available slot templates
+typedef enum{
+    SLOT_10ms,
+    SLOT_20ms_24GHZ,
+    SLOT_40ms_24GHZ,
+    SLOT_40ms_FSK_SUBGHZ,
+    SLOT_40ms_OFDM1MCS0_3_SUBGHZ,
+    MAX_SLOT_TYPES,
+} slotType_t;

```

Comment:
I don't understand why the slot duration is in the name. Just have 3 timeslot templates, for the 3 radio settings we care about.

```
// available slot templates
typedef enum{
    TIMESLOT_TEMPLATE_OQPSK_24GHZ,
    TIMESLOT_TEMPLATE_2FSK_SUBGHZ,
    TIMESLOT_TEMPLATE_OFDM_SUBGHZ,
    TIMESLOT_TEMPLATE_MAX
} timeslot_templates_t;
```

Reply:
There is also a 10ms slot template and 20ms slot template. Thoe former is for python the latter has been classically used for 2.4GHz).